platform/o.n.bootstrap/src/org/netbeans/NbClipboard.java

```diff
@@ -435,20 +436,22 @@ public void run() {
                 // that is used because accessing the clipboard can block
                 // indefinitely. Running the access loop here is deemed similar
                 // in nature.
-                final int MAX_TRIES = 50;
+                final int MAX_TRIES = 10;
                 final long start = System.currentTimeMillis();
+                int delay = 20;
                 for (int i = 0; i < MAX_TRIES; i++) {
                     try {
                         transferable = systemClipboard.getContents(this);
                         break;
                     } catch (IllegalStateException ex) {
                         // Throw exception if retries failed
-                        if (i == (MAX_TRIES - 1) || (System.currentTimeMillis() - start) > 980L) {
+                        if (i == (MAX_TRIES - 1) || (System.currentTimeMillis() + delay - start) > 1000L) {
                             throw ex;
                         } else {
-                            log.log(Level.INFO, "systemClipboard#getContents threw IllegalStateException (try: {0})", i + 1); // NOI18N
+                            log.log(Level.INFO, "systemClipboard#getContents ISE, attempt {0}", i + 1); // NOI18N
                         }
-                        Thread.sleep(20); // Give system time to settle
+                        Thread.sleep(delay); // Give system time to settle
+                        delay *= 2;
```

Comment:
Am I missing something here?!  If the delay is multiplied by two each iteration, how do we get anywhere near `MAX_TRIES` before timing out?

Reply:
Correct, we can never reach MAX_RETRIES. There can be max 7 attempts if getContents fails immediately: 0+20+40+80+160+320+640 = 1260 milliseconds and then the  "> 1000" condition breaks out of the loop the next time it fails). I could get rid of MAX_RETRIES without changing the behavior here.

Reply:
Glad our maths matches!  I would just make sure the values match up in logic so it doesn't confuse the next person - eg. make it `MAX_RETRIES = 7`?  Just in case you hit that one moment the system clock changes and this keeps trying for hours! :laughing:

Reply:
I added the comment "The loop will actually stop before getting to 10 iterations, per the delay formula and conditional throw. But keep the MAX_TRIES just as a fail-safe."